src/screens/mainPage/MainPage.tsx

```diff
@@ -1,23 +1,18 @@
-import {useEffect, useState} from "react"
+import {useEffect, useContext} from "react"
 import CardList from "./components/cardList/CardList"
 import Togglers from "./components/togglers/Togglers"
 import ImageArea from "./components/imageArea/imageArea"
-import {I_Profile} from "../../types/types"
 import Api from "../../api/Api"
-
+import {ProfileContext} from "../../context/profile.context"
 
 const MainPage: React.FC = () => {
-    const [mode, setMode] = useState<string>("Relevant")
-    const [profiles, setProfiles] = useState<I_Profile[]>([])
-
-    console.log("actial data", profiles)
 
+const {profiles, setProfiles, mode} = useContext(ProfileContext)
 
-    const changeMode = (event: React.MouseEvent<HTMLButtonElement>, toggler:string) => {
-    setMode(toggler) 
-    }
+    console.log("actial data", profiles)
 
-    const trackMode = async (mode: string) => {
+    useEffect(() => {
+        const trackMode = async()=>{
         switch (mode){
```

Comment:
Instead of switch case, could use :
`const newMode = await Api.get(...);
setProfiles(newMode);`
also, where does that route going? what are its arguments and what does it receives as response?
"latest" and "stared" are shown twice.

Reply:
latest, stared, relevant are 3 different modes (as string) for 3 different routes (that filter on back). depending on what mode("latest", "relevant", "stared") users clicks, we do api request to 1 of 3 api
at moment only 1 route ("relevant") exists on back (that don't filter all the results from db)